tests/test_vqe.py

```diff
@@ -886,6 +887,38 @@ def circuit(params):
         mt2 = circuit.metric_tensor([p])
         assert np.allclose(mt, mt2)
 
+    def test_multiple_devices(self, mocker):
+        """Test that passing multiple devices to ExpvalCost works correctly"""
```

Comment:
Good to have a test of this behavour now! Surprised we didn't have one before.

I suppose it's another case where 100% coverage != all test cases being tested

Reply:
Yes, luckily one of our demos acted as a test :grin:

Reply:
Yep, the QML demo is actually great in finding edge cases/bugs!